synapse/http/client.py

```diff
@@ -799,6 +755,75 @@ async def get_file(
         )
 
 
+class SimpleHttpClient(BaseSynapseClient):
+    """
+    An HTTP client capable of crossing a proxy and respecting a block/allow list. This
+    does set up a HTTPConnectionPool based on a multiplier of 100 times
+    hs.config.caches.global_factor, as it is responsible for pushing to Sygnal.
+
+    Args:
+        hs: The HomeServer instance to pass in
+        treq_args: Extra keyword arguments to be given to treq.request.
+        ip_blacklist: The IP addresses that are blacklisted that
+            we may not request.
+        ip_whitelist: The whitelisted IP addresses, that we can
+           request if it were otherwise caught in a blacklist.
+        use_proxy: Whether proxy settings should be discovered and used
+            from conventional environment variables.
+    """
+
+    def __init__(
+        self,
+        hs: "HomeServer",
+        treq_args: Optional[Dict[str, Any]] = None,
+        ip_whitelist: Optional[IPSet] = None,
+        ip_blacklist: Optional[IPSet] = None,
+        use_proxy: bool = False,
+    ):
+        super().__init__(hs, treq_args=treq_args)
+        self._ip_whitelist = ip_whitelist
+        self._ip_blacklist = ip_blacklist
+
+        if self._ip_blacklist:
+            # If we have an IP blacklist, we need to use a DNS resolver which
+            # filters out blacklisted IP addresses, to prevent DNS rebinding.
+            self.reactor: ISynapseReactor = BlacklistingReactorWrapper(
+                hs.get_reactor(), self._ip_whitelist, self._ip_blacklist
+            )
+        else:
+            # This should have already been set up in the call to super(). Make sure.
+            self.reactor = hs.get_reactor()
```

Comment:
As you say, this is already set in the reactor, I think it is more confusing to set it again!

```suggestion
        if self._ip_blacklist:
            # If we have an IP blacklist, we need to use a DNS resolver which
            # filters out blacklisted IP addresses, to prevent DNS rebinding.
            self.reactor: ISynapseReactor = BlacklistingReactorWrapper(
                self.reactor, self._ip_whitelist, self._ip_blacklist
            )
```

Reply:
This was a relic left from original code. Pycharm actually mutated this into a `pass`. I'm fine with leaving it out as long as CI is happy.